front_end/scripts/control-and-camera.js

```diff
@@ -0,0 +1,204 @@
+        }
+        manual[e.key] = true;
+    }
+    if (e.key == 's') {
+        client.publish('/smartcar/control/throttle', '-100');
+    }
+    if (e.key == 'a') {
+        client.publish('/smartcar/control/steering', '-45');
+    }
+    if (e.key == 'd') {
+        client.publish('/smartcar/control/steering', '45');
+    }
+})
+
+background.addEventListener('keyup', e => {
+    if (e.key == 'c') {
+        if (!cruiseControl) {
+            cruiseControl = true;
+        } else {
+            cruiseControl = false;
+            client.publish('/smartcar/control/throttle', '0');
+        }
+    }
+    if (e.key == 'm') {
+        manualShift = true;
+        currentGear = 1; //The car starts at first gear
+    }
+    if (e.key == 'n') {
+        manualShift = false;
+    }
+    if (e.key == 'Shift' && manualShift && manual['w']) {
+        if (currentGear < maxGear) {
+            currentGear++;
+            gearShiftAudio();
+        }
+        client.publish('/smartcar/control/throttle', String(currentGear * throttlePerGear));
+    }
+    if (e.key == 'Control' && manualShift && manual['w']) {
+        if (currentGear > 1) {   //Limits the minimum gear to 1
+            currentGear--;
+            gearShiftAudio();
+        }
+        client.publish('/smartcar/control/throttle', String(currentGear * throttlePerGear));
+    }
+    if (e.key == 'w' && !cruiseControl) {
+        client.publish('/smartcar/control/throttle', '0');
+        manual[e.key] = false;
+    }
+    if (e.key == 's') {
+        client.publish('/smartcar/control/throttle', '0');
+        cruiseControl = false;
+    }
+    if (e.key == 'a') {
+        client.publish('/smartcar/control/steering', '0');
+    }
+    if (e.key == 'd') {
+        client.publish('/smartcar/control/steering', '0');
+    }
+})
+
+function gearShiftAudio() {
+    let getGearAudio = "sfx_gearShift" + currentGear;
+    let sfx_gearShift = document.getElementById(getGearAudio);
+    sfx_gearShift.volume = 0.15;
+    sfx_gearShift.play();
+}
+
+function startGamepad(fps) {  //limits the output speed
+    fpsInterval = 1000 / fps;
+    then = Date.now();
+    startTime = then;
+    updateGamepad();
+}
+
+function updateGamepad() {
+    requestAnimationFrame(updateGamepad);
+    now = Date.now();
+    elapsed = now - then;
+    if (elapsed > fpsInterval) {
+        then = now - (elapsed % fpsInterval);
+        let gp = navigator.getGamepads()[0];
+        if (gp !== null) {
+            gp.buttons.forEach(button => {  //curently not used 
+                if (button.pressed == true) {
+                    client.publish('/smartcar/control/buttons', String(gp.buttons.indexOf(button)));
+                }
+            })
+            if (gp.axes[2] >= 0.2 || gp.axes[2] <= -0.2) {
+                client.publish('/smartcar/control/steering', String(gp.axes[2] * 45));
+                x = true;
+            } else if ((gp.axes[2] < 0.2 || gp.axes[2] > -0.2) && x == true) {
```

Comment:
Refer to my first comment for the reasoning.
```suggestion
            } else if ((gp.axes[2] < 0.2 || gp.axes[2] > -0.2) && x) {
```